addons/account/models/company.py

```diff
@@ -171,6 +171,12 @@ class ResCompany(models.Model):
         help="Account that will be set on lines created in cash basis journal entry and used to keep track of the "
              "tax base amount.")
 
+    # Analytics
+    analytic_plan_id = fields.Many2one('account.analytic.plan', string="Default Plan", check_company=True,
+                                       readonly=False,
+                                      default=lambda self: self.env.ref('analytic.analytic_plan_default').id,
+                                               help="Default Plan for a new analytic account")
```

Comment:
What about you create a default here if it is not provided (and no other plan exists for this company) instead of having it in the data?

Reply:
I think it's ok here...
but , @gawa-odoo you said this was required but the setting isn't required?

Reply:
So, plan_id is required on analytic_accounts. We generate analytic_accounts for projects even if they can't see it. So, we need a plan that would be the default for these plans.
We also want to be able to change the default that is proposed for these plans, this is the setting that lets us change the default on the accounts (but only available when you install account, so in the analytic module I can't always rely on this setting)
Now, I can change the way the default plan is generated if we prefer it not to be in the data